ledger/eval.go

```diff
@@ -905,7 +905,7 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 	}
 
 	// Remember this txn
-	cow.addTx(txn.Txn, txid)
+	cow.addTx(txn.Txn, txid, uint64(len(applyData.EvalDelta.InnerTxns))) // Will need to be recursive when inners are
```

Comment:
As a stronger reminder for us to change this if/when inner txns can create more inner txns, I'd recommend checking and returning an error if any inner txn has an inner txn.

Reply:
I'll actually probably write the recursive function, it'll just always bottom out fast.

Reply:
I think, given my last comment to Pavel, that I'm probably wrong about how this is will work with trees of inner transactions.  It won't need to be recursive, exactly, since the inners will already have their children committed to the cb that I'll inevitably need to evaluate them - I'll get their count from there, instead of assuming each = 1 as I do here.  So for now, I will change the comment to remind myself to look at it, but I will not change it to walk down a potential tree of inners.

Reply:
And now that I understand what Pavel was saying, I'll actually be removing the passing of the extra length.